packages/react-components/src/components/TextFieldWithMention/mentionTagUtils.ts

```diff
@@ -305,11 +304,15 @@ type UpdateHTMLProps = {
  * @returns Updated HTML and selection index if the selection index should be set.
  */
 export const updateHTML = (props: UpdateHTMLProps): { updatedHTML: string; updatedSelectionIndex?: number } => {
-  const { htmlText, oldPlainText, newPlainText, tags, startIndex, oldPlainTextEndIndex, change, mentionTrigger } =
-    props;
+  const { htmlText, oldPlainText, tags, startIndex, oldPlainTextEndIndex, change, mentionTrigger } = props;
   if (tags.length === 0 || (startIndex === 0 && oldPlainTextEndIndex === oldPlainText.length - 1)) {
     // no tags added yet or the whole text is changed
-    return { updatedHTML: newPlainText, updatedSelectionIndex: undefined };
+    const changeWithSkippedChars = escapeHTMLChars(change);
+    const updatedHTML =
+      escapeHTMLChars(oldPlainText.substring(0, startIndex)) +
+      changeWithSkippedChars +
+      oldPlainText.substring(oldPlainTextEndIndex);
```

Comment:
We should use escaped version for `substring(oldPlainTextEndIndex)` too or use substrings from `htmlText` instead
Otherwise, in case of `hi < [change] <3` where [change] is `change` value, the second `<` won't be escaped